search/search_integration_tests/pre_ranker_test.cpp

```diff
@@ -0,0 +1,170 @@
+#include "testing/testing.hpp"
+
+#include "search/intermediate_result.hpp"
+#include "search/model.hpp"
+#include "search/pre_ranker.hpp"
+#include "search/ranker.hpp"
+#include "search/search_integration_tests/helpers.hpp"
+#include "search/search_tests_support/test_search_engine.hpp"
+#include "search/suggest.hpp"
+
+#include "indexer/categories_holder.hpp"
+#include "indexer/feature_algo.hpp"
+#include "indexer/features_vector.hpp"
+#include "indexer/mwm_set.hpp"
+#include "indexer/scales.hpp"
+
+#include "generator/generator_tests_support/test_feature.hpp"
+#include "generator/generator_tests_support/test_mwm_builder.hpp"
+
+#include "geometry/mercator.hpp"
+
+#include "platform/country_defines.hpp"
+#include "platform/local_country_file.hpp"
+
+#include "base/assert.hpp"
+#include "base/cancellable.hpp"
+#include "base/limited_priority_queue.hpp"
+#include "base/math.hpp"
+#include "base/stl_add.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/iterator.hpp"
+#include "std/vector.hpp"
+
+using namespace generator::tests_support;
+using namespace search::tests_support;
+
+namespace search
+{
+namespace
+{
+// Returns the boundary such that at least |size| elements from
+// |distance| are less or equal than the boundary.
+double GetBoundary(vector<double> const & distances, size_t size)
+{
+  my::limited_priority_queue<double> queue(size);
+  for (auto const & distance : distances)
+    queue.push(distance);
+  return queue.empty() ? 0.0 : queue.top();
+}
+
+class FakeRanker : public Ranker
```

Comment:
mpimenov <notifications@github.com> writes:

> In search/search_integration_tests/pre_ranker_test.cpp:
>
>> +
>> +namespace search
>> +{
>> +namespace
>> +{
>> +// Returns the boundary such that at least |size| elements from
>> +// |distance| are less or equal than the boundary.
>> +double GetBoundary(vector<double> const & distances, size_t size)
>> +{
>> +  my::limited_priority_queue<double> queue(size);
>> +  for (auto const & distance : distances)
>> +    queue.push(distance);
>> +  return queue.empty() ? 0.0 : queue.top();
>> +}
>> +
>> +class FakeRanker : public Ranker
>
> s/FakeRanker/TestRanker/ ?
Real difference?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.*
>

-- 
Best regards, Yuri